From b4dc22cab7b8b071c8a150b9c61edaa247bacb6a Mon Sep 17 00:00:00 2001
From: Harald Hoyer <harald@redhat.com>
Date: Thu, 18 Apr 2013 12:54:55 +0200
Subject: [PATCH] dracut-install: error out, if ldd reports no execution
 permission

This turns off lazy resolving on noexec mounted tmp directories.

https://bugzilla.redhat.com/show_bug.cgi?id=953426
---
 install/dracut-install.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/install/dracut-install.c b/install/dracut-install.c
index 5040dea..2d0412c 100644
--- a/install/dracut-install.c
+++ b/install/dracut-install.c
@@ -285,6 +285,12 @@ static int resolve_deps(const char *src)
 
                 log_debug("ldd: '%s'", buf);
 
+                if (strstr(buf, "you do not have execution permission")) {
+                        log_error(buf);
+                        ret+=1;
+                        break;
+                }
+
                 if (strstr(buf, "not a dynamic executable"))
                         break;
 
